assets/js/phoenix_live_view.js

```diff
@@ -1568,6 +1563,21 @@ export class View {
     this.liveSocket.log(this, kind, msgCallback)
   }
 
+  withinTargets(phxTarget, callback){
+    if(/^(0|[1-9](\d?)+)$/.test(phxTarget)){
+      let targets = DOM.findComponentNodeList(this.el, phxTarget)
+      if(targets.length === 0){
+        logError(`no phx-target's found matching @myself of ${phxTarget}`)
+      } else {
+        callback(targets[0])
+      }
+    } else {
+      let targets = Array.from(this.el.querySelectorAll(phxTarget))
+      if(targets.length === 0){ logError(`no phx-target's found for selector "${phxTarget}"`) }
```

Comment:
I changed this to `logError` because race conditions may actually mean that nothing will be found.